proxy/envoy/config.go

```diff
@@ -549,8 +549,7 @@ func buildClusters(services []*model.Service) []Cluster {
 				LbType:           DefaultLbType,
 				ConnectTimeoutMs: DefaultTimeoutMs,
 			}
-			if port.Protocol == model.ProtocolGRPC ||
-				port.Protocol == model.ProtocolHTTP2 {
+			if port.Protocol != model.ProtocolTCP && port.Protocol != model.ProtocolUDP {
```

Comment:
Agreed. But i think this code will still not work. Because we cannot distinguish between egress service clusters and non-egress clusters. For clusters inside k8s/mesos, i.e. within istio deployment, we have Envoy end to end, i.e., we can do http2 inside istio. But when going out, we cannot assume anything. So, we need to stick to http1.